Add report feature, add random extensions

rzerdik opened the pull request:
fixes #15 

added report creation feature, cleanup

Commit:
Add report feature, add random extensions

Commit:
Add download to report dialog

fpindej commented:
The functionality works on my end. I hope I tested everything 😄 

As for my take on this one: I wouldn't burn any more time and would ship it, since this is basically the entire functionality we need covered (afaik).

fpindej commented on src/Client/Extensions/JSRuntimeExtensions.cs:
```diff
@@ -0,0 +1,12 @@
+{
+    public async static Task SaveAsAsync(this IJSRuntime js, string fileName, byte[] data)
+    {
+        await js.InvokeAsync<object>("saveAsFile", fileName, Convert.ToBase64String(data));
```
Expression body method

petrspelos commented on src/Client/Dialogs/ReportDialog.razor:
```diff
@@ -0,0 +1,9 @@
+        <MudButton Color="Color.Default" OnClick="Cancel">Close</MudButton>
+        <MudButton Color="Color.Primary" OnClick="Download">Download</MudButton>
+    </DialogActions>
+</MudDialog>
```
```suggestion
</MudDialog>

```

petrspelos commented on src/Client/Extensions/RandomExtensions.cs:
```diff
@@ -0,0 +1,17 @@
+
+        return list.Count == 0 ? default! : list[random.Next(0, list.Count)];
+    }
+}
```
```suggestion
}

```

petrspelos commented on src/Client/wwwroot/js/saveAsFile.js:
```diff
@@ -0,0 +1,9 @@
+    document.body.appendChild(link); // Needed for Firefox
+    link.click();
+    document.body.removeChild(link);
+}
```
```suggestion
}

```

petrspelos reviewed (changes requested):
Nothing obviously wrong with it. ☺️ 
Just a couple of missing newline. I also agree with @fpindej's expression body suggestion.

Let's make these changes and get this thing in! 💯 🎉

fpindej commented:
@thezerda If You don't mind, I will commit @petrspelos suggestions and will get this into production asap 😊 Gotta try the new deployment pipeline!

fpindej committed:
Update src/Client/Dialogs/ReportDialog.razor

Co-authored-by: Petr Sedláček <petrspelos@users.noreply.github.com>

fpindej committed:
Update src/Client/Extensions/RandomExtensions.cs

Co-authored-by: Petr Sedláček <petrspelos@users.noreply.github.com>

fpindej committed:
Update src/Client/wwwroot/js/saveAsFile.js

Co-authored-by: Petr Sedláček <petrspelos@users.noreply.github.com>

fpindej committed:
Introduce expression body method

Introduce expression body methon on SaveAsAsync method in JSRuntimeExntensions class

fpindej commented:
@petrspelos Re-review please 🕵🏻
